### PR TITLE
matrix: add canonical data

### LINKED ISSUE #569: matrix: Implement canonical-data.json
We want to have a standard set of test inputs and outputs for each exercise to
make it easier to port them to new languages, as well as to help keep the
exercises in sync.

The Matrix exercise can be found in the [./exercises/matrix/][exercise-dir] directory.

The step-by-step instructions for how to compile a canonical-data.json file is
described in [this section of the contributing guide][contributing].

See http://exercism.io/contribute/canonical-data/matrix for the up-to-date list of
language tracks that have an implementation of the Matrix exercise.

The [tracking issue][tracking-issue] has more context.

[contributing]: https://github.com/exercism/docs/blob/master/you-can-help/improve-exercise-metadata.md#extracting-canonical-test-data
[tracking-issue]: https://github.com/exercism/x-common/issues/552
[exercise-dir]: https://github.com/exercism/x-common/tree/master/exercises/matrix

### PR DESCRIPTION
**Pull request by Ciamar**
Resolves #569 

I have some doubts about what I did, particularly about the use of an index as an input... in the problem specification it says nothing about extracting one individual row/column, but most implementations tested for that, so I tried to make the test cases that way. 

Also, I wasn't sure how many test cases would be reasonable for this exercise.

[description.md](https://github.com/exercism/problem-specifications/blob/master/exercises/matrix/description.md)

### CODE REVIEW

File: exercises/matrix/canonical-data.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "exercise": "matrix",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "one number matrix",
+      "property": "rows",
```
Comment (ErikSchierboom) [inline exercises/matrix/canonical-data.json]: Seeing how only a single index is being used, shouldn't this be named `row` (singular)?

File: exercises/matrix/canonical-data.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      "input": {
+        "string": "1 2\n10 20",
+        "index": 1
+      },
+      "expected": [10,20]
+    },
+    {
+      "description": "can extract column",
+      "property": "columns",
```
Comment (ErikSchierboom) [inline exercises/matrix/canonical-data.json]: Seeing how only a single index is being used, shouldn't this be named `column` (singular)?

**Comment by Ciamar** [comment]
I agree that it makes more sense to have the function names in singular, and I added the suggested test cases. 

I made a bit of a mess with the commits because I forgot a comma and tried to amend, but I think is fine now.

File: exercises/matrix/canonical-data.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    {
+      "description": "extract column where numbers have different widths",
+      "property": "column",
+      "input": {
+        "string": "89 1903 3\n18 3 1\n9 4 800",
+        "index": 1
+      },
+      "expected": [1903, 3, 4]
+    }
+    ]
```
Comment (ErikSchierboom) [inline exercises/matrix/canonical-data.json]: very minor nit: this should be indented one level to the level I think.

File: exercises/matrix/canonical-data.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      "expected": [10,20]
+    },
+    {
+      "description": "can extract row from non-square matrix",
+      "property": "row",
+      "input": {
+        "string": "1 2 3\n4 5 6\n7 8 9\n8 7 6",
+        "index": 2
+      },
+      "expected": [4, 5, 6]
```
Comment (petertseng) [inline exercises/matrix/canonical-data.json]: index 2 is 7 8 9 not 4 5 6

Comment (petertseng) [inline exercises/matrix/canonical-data.json]: index 2 is 7 8 9 not 4 5 6

File: exercises/matrix/canonical-data.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
+{
+  "exercise": "matrix",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "extract row from one number matrix",
+      "property": "row",
+      "input": {
+        "string": "1",
```
Comment (ErikSchierboom) [inline exercises/matrix/canonical-data.json]: I was thinking, maybe `matrix` would be a better name than `string`?

File: exercises/matrix/canonical-data.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      "expected": [1]
+    },
+    {
+      "description": "can extract row",
+      "property": "row",
+      "input": {
+        "string": "1 2\n3 4",
+        "index": 1
+      },
+      "expected": [3,4]
```
Comment (petertseng) [inline exercises/matrix/canonical-data.json]: this line is missing the space after its comma (and the space is present in other lines)

File: exercises/matrix/canonical-data.json [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    {
+      "description": "extract row where numbers have different widths",
+      "property": "row",
+      "input": {
+        "string": "1 2\n10 20",
+        "index": 1
+      },
+      "expected": [10,20]
```
Comment (petertseng) [inline exercises/matrix/canonical-data.json]: this line is missing the space after its comma (and the space is present in other lines)

File: exercises/matrix/canonical-data.json (hunk 6 above)
Comment (Ciamar) [reply exercises/matrix/canonical-data.json]: I'm not sure what you mean, I don't see a space after the comma in any other line.

Comment (petertseng) [reply exercises/matrix/canonical-data.json]: line 39. 

`"expected": [7, 8, 9]`

line 57.

`"expected": [3, 6, 9]`

File: exercises/matrix/canonical-data.json [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    {
+      "description": "can extract column",
+      "property": "column",
+      "input": {
+        "string": "1 2 3\n4 5 6\n7 8 9",
+        "index": 2
+      },
+      "expected": [3, 6, 9]
+
```
Comment (petertseng) [inline exercises/matrix/canonical-data.json]: can remove this empty line

**Comment by ErikSchierboom** [comment]
Squashed and merged! Thanks @Ciamar!

**Pull request merged**